File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPAdView.java
package com.monet.bidder;

import android.content.Context;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

class DFPAdView implements AdServerAdView {
  private final com.google.android.gms.ads.AdView mAdView;
  private String mAdUnitId;
  private Context mContext;

  DFPAdView(com.google.android.gms.ads.AdView adView) {
    mAdView = adView;
    mContext = adView.getContext();
    mAdUnitId = adView.getAdUnitId();
  }

  @Override
  public AdServerWrapper.Type getType() {
    return AdServerWrapper.Type.BANNER;
  }

  @Override
  public String getAdUnitId() {
    return mAdUnitId;
  }

  @Override
  public void setAdUnitId(String adUnitId) {
    mAdUnitId = adUnitId;
  }

  @Override
  public Context getContext() {
    return mContext;
  }

  @Override
  public void loadAd(AdServerAdRequest request) {
    DFPAdViewRequest adRequest = (DFPAdViewRequest) request;
    if (mAdView != null) {
      mAdView.loadAd(adRequest.getDFPRequest());
    }
  }
}

File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPAdView.kt
package com.monet.bidder

import android.annotation.SuppressLint
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.monet.AdServerAdRequest
import com.monet.AdServerAdView
import com.monet.AdType
import com.monet.AdType.BANNER
import com.monet.DFPAdViewRequest

class DFPAdView(private val adView: AdView?) : AdServerAdView {
  override var adUnitId: String = adView?.adUnitId ?: ""
  override var type: AdType = BANNER

  @SuppressLint("MissingPermission")
  override fun loadAd(request: AdServerAdRequest?) {
    request?.let {
      val adRequest = it as DFPAdViewRequest
      adView?.loadAd(adRequest.adRequest.request as AdRequest)
    }
  }
}